addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputWithDabControlXML.java

```diff
@@ -0,0 +1,236 @@
+        } else {
+            Node playInfoNode = XMLUtils.getNode(doc.getFirstChild(), "DAB/Play_Info/" + msgForBand.band);
+
+            msgForPreset.presetChannel = XMLUtils.getNodeContentOrDefault(playInfoNode, "Preset/Preset_Sel", -1);
+            logger.debug("Preset set to {} for input {}", msgForPreset.presetChannel, inputID);
+
+            Node metaInfoNode = XMLUtils.getNode(playInfoNode, "Meta_Info");
+            if (metaInfoNode != null) {
+                msgForPlayInfo.album = XMLUtils.getNodeContentOrDefault(metaInfoNode, "Program_Type", "");
+                if (BAND_FM.equals(msgForBand.band)) {
+                    msgForPlayInfo.station = XMLUtils.getNodeContentOrDefault(metaInfoNode, "Program_Service", "");
+                    msgForPlayInfo.artist = XMLUtils.getNodeContentOrDefault(metaInfoNode, "Station", "");
+                    msgForPlayInfo.song = XMLUtils.getNodeContentOrDefault(metaInfoNode, "Radio_Text", "");
+                } else if (BAND_DAB.equals(msgForBand.band)) {
+                    msgForPlayInfo.station = XMLUtils.getNodeContentOrDefault(metaInfoNode, "Service_Label", "");
+                    msgForPlayInfo.artist = XMLUtils.getNodeContentOrDefault(metaInfoNode, "Ensemble_Label", "");
+                    msgForPlayInfo.song = XMLUtils.getNodeContentOrDefault(metaInfoNode, "DLS", "");
+                }
+            }
+        }
+
+        // DAB does not provide channel names, the channel list will be empty
+        msgForPreset.presetChannelNamesChanged = true;
+        msgForPreset.presetChannelNames = new String[0];
+
+        if (observerForBand != null) {
+            observerForBand.dabBandUpdated(msgForBand);
+        }
+        if (observerForPreset != null) {
+            observerForPreset.presetInfoUpdated(msgForPreset);
+        }
+        if (observerForPlayInfo != null) {
+            observerForPlayInfo.playInfoUpdated(msgForPlayInfo);
+        }
+    }
+
+    @Override
+    public void selectBandByName(String band) throws IOException, ReceivedMessageParseException {
+        // Example: <Play_Control><Band>FM</Band></Play_Control>
+        String cmd = String.format("<Play_Control><Band>%s</Band></Play_Control>", band);
+        comReference.get().send(wrInput(cmd));
+        update();
+    }
+
+    @Override
+    public void selectItemByPresetNumber(int presetChannel) throws IOException, ReceivedMessageParseException {
+        if (bandState == null || bandState.band == null || bandState.band.isEmpty()) {
+            logger.warn("Cannot change preset because the band is unknown for input {}", inputID);
+            return;
+        }
+
+        // Example: <Play_Control><FM><Preset><Preset_Sel>2</Preset_Sel></Preset></FM></Play_Control>
+        String cmd = String.format("<Play_Control><%s><Preset><Preset_Sel>%d</Preset_Sel></Preset></%s></Play_Control>",
```

Comment:
I thought this PR was also about adding JSON support or does it live side-by-side now?